master/internal/config.go

```diff
@@ -102,6 +106,35 @@ func (c Config) Printable() ([]byte, error) {
 	return optJSON, nil
 }
 
+// Resolve resolves the values in the configuration.
+func (c *Config) Resolve() error {
+	if c.Port == 0 {
+		if c.Security.TLS.Enabled() {
+			c.Port = 8443
+		} else {
+			c.Port = 8080
+		}
+	}
+
+	root, err := filepath.Abs(c.Root)
+	if err != nil {
+		return err
+	}
+	c.Root = root
+
+	c.DB.Migrations = fmt.Sprintf("file://%s", filepath.Join(c.Root, "static/migrations"))
+
+	c.ResourceManager, c.ResourcePoolsConfig, err = resourcemanagers.ResolveConfig(
```

Comment:
yes just to maintain backward compatibility internally because we need to be able to configure resource manager and pools for dev purposes. We might not maintain the backward compatibility for users when we release this feature.